src/vs/platform/terminal/node/ptyService.ts

```diff
@@ -461,9 +464,13 @@ export class PersistentTerminalProcess extends Disposable {
 		reviveBuffer: string | undefined,
 		private _icon?: TerminalIcon,
 		private _color?: string,
+		name?: string,
 		fixedDimensions?: IFixedTerminalDimensions
 	) {
 		super();
+		if (name) {
+			this.setTitle(name, TitleEventSource.Api);
```

Comment:
this should work, but attachPersistentProcess.title is currently ''
https://github.com/microsoft/vscode/blob/aed2436349622a39ac1593cdad3236149897cb3f/src/vs/workbench/contrib/terminal/browser/terminalInstance.ts#L362